flyteadmin/auth/auth_context.go

```diff
@@ -207,8 +207,10 @@ func GetOAuth2ClientConfig(ctx context.Context, options config.OpenIDOptions, pr
 
 	secret = strings.TrimSuffix(secret, "\n")
 
+	logger.Infof(ctx, "FOO %v %v", URLFromContext(ctx), URLFromContext(ctx).ResolveReference(callbackRelativeURL).String())
+
 	return oauth2.Config{
-		RedirectURL:  callbackRelativeURL.String(),
+		RedirectURL:  URLFromContext(ctx).ResolveReference(callbackRelativeURL).String(),
```

Comment:
I believe the previous logic `RedirectURL: callbackRelativeURL.String()` causes issues when a deployment is using a custom base URL like `/custom-flyte/console` rather than `/console`. This is because the generated redirectURL for callbacks in the auth flow then becomes `redirect=host/callback` rather than `redirect=host/custom-flyte/console` which causes the callback to fail.

The fix here is to mirror how its done here in OAuth2ClientConfig https://github.com/ddl-rliu/flyte/blob/7953d627adf73dc66089f8d30e23acb594792bee/flyteadmin/auth/auth_context.go#L63

Here it resolves the callback using the request URL, to get an absolute URL that can be happily used in the auth flow.

Reply:
Update: my reasoning here is likely incorrect, still investigating